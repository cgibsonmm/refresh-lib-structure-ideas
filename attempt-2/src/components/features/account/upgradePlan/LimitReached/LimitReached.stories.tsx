import { StoryObj, Meta } from '@storybook/react';
import { LimitReached } from './LimitReached';

const meta: Meta<typeof LimitReached> = {
  title: 'Components/UpgradePlan/LimitReached',
  component: LimitReached,
  argTypes: {
    onContactUs: { action: 'contact us' },
  },
};

export default meta;

type Story = StoryObj<typeof LimitReached>;

export const Default: Story = {};
