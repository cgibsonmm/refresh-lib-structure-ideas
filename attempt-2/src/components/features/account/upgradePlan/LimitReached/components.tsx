import { Box, styled } from '@/theme/index';

export const LimitReachedContainer = styled(Box)(({ theme }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  flexDirection: 'column',

  [theme.breakpoints.up('lg')]: {
    flexDirection: 'row',
  },
}));

export const LimitReachedDescription = styled(Box)(({ theme }) => ({
  paddingTop: theme.spacing(4),
  paddingLeft: theme.spacing(4),
  paddingRight: theme.spacing(4),

  [theme.breakpoints.up('sm')]: {
    paddingTop: theme.spacing(5),
    paddingLeft: theme.spacing(5),
    paddingRight: theme.spacing(5),
  },

  [theme.breakpoints.up('md')]: {
    paddingTop: theme.spacing(7),
    paddingLeft: theme.spacing(7),
    paddingRight: theme.spacing(7),
  },

  [theme.breakpoints.up('lg')]: {
    paddingTop: theme.spacing(11),
    paddingLeft: theme.spacing(10),
  },
}));

export const LimitReachedHeader = styled(Box)(({ theme }) => ({
  marginBottom: theme.spacing(4),

  [theme.breakpoints.up('md')]: {
    width: 614,
  },
}));

export const LimitReachedImageContainer = styled(Box)(({ theme }) => ({
  maxWidth: 370,
  alignSelf: 'end',
  overflow: 'hidden',

  [theme.breakpoints.up(410)]: {
    maxWidth: 400,
  },

  [theme.breakpoints.up(768)]: {
    maxWidth: 750,
  },

  [theme.breakpoints.up('lg')]: {
    maxWidth: 860,
  },

  '& img': {
    objectFit: 'cover',
    width: 550,

    [theme.breakpoints.up(768)]: {
      width: 1000,
      height: 520,
    },

    [theme.breakpoints.up('lg')]: {
      width: 'auto',
      height: 'auto',
    },
  },
}));
