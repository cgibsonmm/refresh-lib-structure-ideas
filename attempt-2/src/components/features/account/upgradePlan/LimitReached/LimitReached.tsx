import React, { useContext } from 'react';
import { AppConstants } from '@/context/AppConstants';
import { Box } from '@/theme/index';
import {
  LimitReachedContainer,
  LimitReachedDescription,
  LimitReachedHeader,
  LimitReachedImageContainer,
} from './components';
import { CustomSolutionButton, Heading, SubHeading } from '../components';

export interface LimitReachedProps {
  /** Callback fired when the user clicks the **Contact us now** button. */
  onContactUs: () => void;
}

/**
 * This component is used to display the limit reached message in the upgrade plan flow.
 * This happens when the user has already upgraded to the highest plan available and we don't
 * have any more plans to offer.
 */
export function LimitReached({ onContactUs }: LimitReachedProps) {
  const { imgComputerLimitReached } = useContext(AppConstants);

  return (
    <LimitReachedContainer className="limit-reached">
      <Box>
        <LimitReachedDescription>
          <LimitReachedHeader>
            <Heading component="p" variant="h1">
              Power through your limit?
            </Heading>

            <Heading component="p" variant="h1" sx={{ fontWeight: 'bold' }}>
              Reach out to keep the momentum going!
            </Heading>
          </LimitReachedHeader>

          <SubHeading variant="h2">
            We offer customized solutions to match your needs
          </SubHeading>

          <CustomSolutionButton
            variant="contained"
            color="success"
            size="large"
            onClick={onContactUs}
          >
            Contact us now
          </CustomSolutionButton>
        </LimitReachedDescription>
      </Box>

      <LimitReachedImageContainer>
        <img src={imgComputerLimitReached} alt="Limit reached" />
      </LimitReachedImageContainer>
    </LimitReachedContainer>
  );
}
