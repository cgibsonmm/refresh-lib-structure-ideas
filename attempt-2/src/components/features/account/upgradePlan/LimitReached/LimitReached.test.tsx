import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LimitReached, LimitReachedProps } from './LimitReached';

const props: LimitReachedProps = {
  onContactUs: jest.fn(),
};

test('renders the correct content', () => {
  const { asFragment } = render(<LimitReached {...props} />);

  expect(asFragment()).toMatchSnapshot();
});

test('contact us', async () => {
  render(<LimitReached {...props} />);

  const contactUs = screen.getByRole('button');

  await userEvent.click(contactUs);

  expect(props.onContactUs).toHaveBeenCalled();
});
