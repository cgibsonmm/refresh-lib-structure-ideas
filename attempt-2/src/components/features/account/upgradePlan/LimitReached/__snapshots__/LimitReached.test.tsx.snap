// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct content 1`] = `
<DocumentFragment>
  <div
    class="limit-reached MuiBox-root css-b55jur"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-q2map3"
      >
        <div
          class="MuiBox-root css-rvvazp"
        >
          <p
            class="MuiTypography-root MuiTypography-h1 css-xlu8vr-MuiTypography-root"
          >
            Power through your limit?
          </p>
          <p
            class="MuiTypography-root MuiTypography-h1 css-1s6zrpn-MuiTypography-root"
          >
            Reach out to keep the momentum going!
          </p>
        </div>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-14yvy03-MuiTypography-root"
        >
          We offer customized solutions to match your needs
        </h2>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge css-1sbu2i7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Contact us now
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-8qzsz8"
    >
      <img
        alt="Limit reached"
        src="https://via.placeholder.com/1142x839"
      />
    </div>
  </div>
</DocumentFragment>
`;
